src/System.Linq.Expressions/tests/Dynamic/CallInfoTests.cs

```diff
@@ -12,112 +12,77 @@ namespace System.Dynamic.Tests
     public class CallInfoTests
     {
         [Fact]
-        public void NullNames()
+        public void Ctor_NullNames_ThrowsArgumentNullException()
         {
             Assert.Throws<ArgumentNullException>("argNames", () => new CallInfo(0, default(IEnumerable<string>)));
             Assert.Throws<ArgumentNullException>("argNames", () => new CallInfo(0, default(string[])));
         }
 
-        [Fact]
-        public void NegativeCount()
-        {
-            Assert.Throws<ArgumentException>(null, () => new CallInfo(-1));
-        }
-
-        [Fact]
-        public void CountLessThanNameCount()
+        [Theory]
+        [InlineData(-1, new string[0])]
+        [InlineData(-1, new string[] { "foo", "bar", "baz", "quux", "quuux" })]
```

Comment:
Nit: should the -1 be changed to 2, to match what was there before and to be clear the argument error isn't about the value being, for example, less than 0?